### ISSUE TITLE
[Question] Cortana Skill says "invalid channel data"?

### ISSUE THREAD

**Issue by jsiegmund** [Microsoft/BotBuilder#3173]
## System Information (Required)
* SDK Language: C#
* SDK Version: 3.8.5
* Development Environment: AZURE_BOT_SERVICE

## Issue Description
I created a bot for a Cortana Skill. It all works fine when I run it locally, but once published to an Azure Bot Service instance, I'm getting "invalid channel data" whenever an action is being executed. I have debugged my code remotely only to find that my stuff is working and exiting, the problem seems to lie in the framework somewhere. For more info also refer to https://stackoverflow.com/questions/45161505/debugging-published-bot?noredirect=1#comment77297976_45161505.

## Code Example
Full repo is published at https://github.com/jsiegmund/bots-domoticz

## Steps to Reproduce the Issue
1. Create Cortana Skill
2. Publish to Bot Service instance in Azure
3. Call via Cortana

## Expected Behavior
Would expect it to work in the same way as it does on-prem.

## Actual Results
Bot service instance is showing "invalid channel data" in the event trace. Azure diagnotics showing:

> 2017-07-18T08:11:34.252 Exception while executing function: Functions.messages 
> 2017-07-18T08:11:34.252 Invalid channel data.
> 2017-07-18T08:11:34.272 Executed 'Functions.messages' (Failed, Id=e5eae9f5-97e4-4a35-aa5f-5c7d58a63aed)
> 2017-07-18T08:11:34.272 Invalid channel data.
> 2017-07-18T08:11:34.272 Function had errors. See Azure WebJobs SDK dashboard for details. Instance ID is 'e5eae9f5-97e4-4a35-aa5f-5c7d58a63aed'
> 2017-07-18T08:11:34.272 Invalid channel data.
> 2017-07-18T08:11:34.315 {"id":"6f095c10-d9e8-4ad0-a6a8-caf052288f8b","requestId":"2f4c950c-2459-4467-8579-4f3b5bc45e6b","statusCode":500,"errorCode":0,"messsage":"An error has occurred. For more information, please check the logs for error ID 6f095c10-d9e8-4ad0-a6a8-caf052288f8b"}
> 2017-07-18T08:11:34.315 Invalid channel data.
> 2017-07-18T08:11:34.194 Function completed (Failure, Id=e5eae9f5-97e4-4a35-aa5f-5c7d58a63aed, Duration=4626ms)
> 2017-07-18T08:11:34.252 Exception while executing function: Functions.messages. Microsoft.Bot.Connector: Invalid channel data.

**Comment by jsiegmund** [comment]
I just upgraded everything to framework version 3.9.0. This did not fix the error. What I find the weirdest is that I'm not having these issues when running locally. So is using bot service for this scenario supported or should I switch to a regular webApi setup instead?

**Comment by nwhitmont** [comment]
@jsiegmund Thanks for reporting.  I have passed this on to our Bot Service & Functions team to investigate. Will update when I have more details.

**Comment by nwhitmont** [comment]
@jsiegmund For reference, what is your bot App ID?

**Comment by jsiegmund** [comment]
Sorry, it's been a bit busy. App id = 4f097b1b-b18e-49bf-bc7c-0ed23c56a019

**Comment by nwhitmont** [comment]
Hi @jsiegmund - When I looked at your bot's Cortana configuration, it said that the bot cannot be added to Cortana channel because the email account that was used to register the bot is not a MSA aka personal Microsoft account. 

If you used an AAD account to register the bot, then it won't work. You need to use an existing personal MSA, or [create a new MSA account](https://account.microsoft.com/account), to register the bot and connect it to Cortana channel.

**Comment by jsiegmund** [comment]
Hi @nwhitmont this starts explaining stuff. Actually it is an MSA account which I used, but I recently had to change the alias for it due to some other Azure and Outlook.com related problems. When I did, I ended up in a world of hurt with all sorts of Azure stuff not working any longer. My botservice instances were one of the things that didn't work. So they first shifted my subscriptions to another account and then finally I have changed back the primary alias to what it used to be. Unfortunately it still doesn't seem to work properly cause when I log in to the portal it shows the other alias even though that is now secondary. That being said; both mail addresses should point to the same MSA. I have the feeling that a number of Azure related services do not understand the concept of MSA aliases and certainly not switching them. They should use a MSA unique ID behind the curtains but it seems they not all do and maybe store the e-mail address instead. Can you confirm this?

**Comment by jsiegmund** [comment]
A little bit more research. I recreated the Cortana channel although I doubt a little whether it was actually deleted cause when I tried to recreate it; all the properties were already there filled in with the previous values. 

The weird thing is that when I log into the Cortana Dashboard and check my account (top right); it shows my name and the e-mail address that has been set as the primary e-mail address on my MSA. But when I log into the bot framework dashboard; it shows my name tied to my secondary MSA e-mail address. So that might be causing issues, don't know, but I would not know how to convince the bot portal to take on my primary e-mail instead?
